node/storage-cleaner/src/appwrite.js

```diff
@@ -0,0 +1,39 @@
+import { Client, Storage, Query } from 'node-appwrite';
+import { getExpiryDate } from './util.js';
+
+class AppwriteService {
+  constructor() {
+    const client = new Client()
+      .setEndpoint(
+        process.env.APPWRITE_ENDPOINT ?? 'https://cloud.appwrite.io/v1'
+      )
+      .setProject(process.env.APPWRITE_FUNCTION_PROJECT_ID)
+      .setKey(process.env.APPWRITE_API_KEY);
+    this.storage = new Storage(client);
+  }
+
+  /**
+   * Clean up files from the storage bucket by removing files older than a specified retention period.
+   *
+   * @param {string} bucketId - The ID of the storage bucket to clean.
+   * @returns {Promise<void>} A Promise that resolves when the bucket is cleaned.
+   */
+  async cleanBucket(bucketId) {
+    let hasNextPage = true;
+    const queries = [
+      Query.lessThan('$createdAt', getExpiryDate()),
+      Query.limit(25),
+    ];
+    do {
+      const response = await this.storage.listFiles(bucketId, queries);
+      await Promise.all(
+        response.files.map((file) =>
+          this.storage.deleteFile(bucketId, file.$id)
+        )
+      );
+      hasNextPage = response.files.length > 0;
+    } while (hasNextPage);
```

Comment:
I wonder if we really need the   `  let hasNextPage = true;` var, can we just check it in `response.files.length > 0` ?

I feel like it's readable enough here.

Reply:
One other method can be to use recursion here, we can call the same cleanBucket again if the response.files.length is greater than 0.